### DISCUSSION TITLE
Spawn command does not work as in terminal

### DISCUSSION THREAD

**Discussion by gruvw** [qtile/qtile#3879] [Q&A] [answered]
Hello,

I don't know why, but when I use `lazy.spawn("import png:- | xclip -selection clipboard -t image/png")`, which is supposed to take an area screenshot and copy it to my clipboard, it does not work.  
However, when I simply paste my command in my terminal, it works perfectly.
What makes qtile's spawn different from running a command in my terminal ?  
Does qtile uses a different user ?  

Thanks a lot.

**Comment by elParaguayo** [comment, marked as answer]
Try `lazy.spawn("import png:- | xclip -selection clipboard -t image/png", shell=True)`

**Reply by gruvw** [reply]
Great thank you! Could not find that in the docs :(

Btw, when should I use `shell=True` ?

**Reply by elParaguayo** [reply]
Adding that flag means the command is run in a separate shell (by prefixing your command with `/bin/sh -c`).

The times you may need to do this are as follows (copied from the python docs on the `subprocess` module):
> If shell is True, the specified command will be executed through the shell. This can be useful if you are using Python primarily for the enhanced control flow it offers over most system shells and still want convenient access to other shell features such as shell pipes, filename wildcards, environment variable expansion, and expansion of ~ to a user’s home directory. However, note that Python itself offers implementations of many shell-like features (in particular, [glob](https://docs.python.org/3/library/glob.html#module-glob), [fnmatch](https://docs.python.org/3/library/fnmatch.html#module-fnmatch), [os.walk()](https://docs.python.org/3/library/os.html#os.walk), [os.path.expandvars()](https://docs.python.org/3/library/os.path.html#os.path.expandvars), [os.path.expanduser()](https://docs.python.org/3/library/os.path.html#os.path.expanduser), and [shutil](https://docs.python.org/3/library/shutil.html#module-shutil)).
